airbyte-integrations/connectors/destination-bigquery-denormalized/src/test/java/io/airbyte/integrations/destination/bigquery/BigQueryDenormalizedDestinationTest.java

```diff
@@ -0,0 +1,424 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class BigQueryDenormalizedDestinationTest {
+
+  @Mock
+  UploaderConfig uploaderConfigMock;
+  @Mock
+  ConfiguredAirbyteStream configuredStreamMock;
+  @Mock
+  AirbyteStream airbyteStreamMock;
+  @Mock
+  DefaultBigQueryDenormalizedRecordFormatter bigQueryRecordFormatterMock;
+  @Mock
+  BigQuery bigQueryMock;
+
+  MockedStatic<BigQueryUploaderFactory> uploaderFactoryMock;
+
+  @InjectMocks
+  BigQueryDenormalizedDestination bqdd;
+
+  @BeforeEach
+  void init() {
+    uploaderFactoryMock = Mockito.mockStatic(BigQueryUploaderFactory.class, Mockito.CALLS_REAL_METHODS);
+    uploaderFactoryMock.when(() -> BigQueryUploaderFactory.getUploader(any(UploaderConfig.class))).thenReturn(mock(BigQueryDirectUploader.class));
+  }
+
+  @AfterEach
+  public void teardown() {
+    uploaderFactoryMock.close();
+  }
+
+  @Test
+  void getTargetTableName() {
+    final String streamName = "stream_name";
+    BigQuerySQLNameTransformer nameTransformerMock = mock(BigQuerySQLNameTransformer.class);
+    bqdd.getTargetTableName(streamName);
+    verify(nameTransformerMock, times(0)).getIdentifier(streamName);
+  }
+
+  @Test
+  void getFormatterMap() {
+    final JsonNode jsonNodeSchema = getSchema();
+    final Map<UploaderType, BigQueryRecordFormatter> formatterMap = bqdd.getFormatterMap(jsonNodeSchema);
+    assertEquals(2, formatterMap.size());
+    assertTrue(formatterMap.containsKey(UploaderType.AVRO));
+    assertTrue(formatterMap.containsKey(UploaderType.STANDARD));
+    assertThat(formatterMap.get(UploaderType.AVRO), instanceOf(GcsBigQueryDenormalizedRecordFormatter.class));
+    assertThat(formatterMap.get(UploaderType.STANDARD), instanceOf(DefaultBigQueryDenormalizedRecordFormatter.class));
+  }
+
+  @Test
+  void isDefaultAirbyteTmpTableSchema() {
+    assertFalse(bqdd.isDefaultAirbyteTmpTableSchema());
+  }
+
+  @Test
+  void getAvroSchemaCreator() {
+    assertNotNull(bqdd.getAvroSchemaCreator());
+  }
+
+  @Test
+  void getRecordFormatterCreator() {
+    final BigQuerySQLNameTransformer nameTransformerMock = mock(BigQuerySQLNameTransformer.class);
+    final BigQueryRecordFormatter resultFormatter = bqdd.getRecordFormatterCreator(nameTransformerMock)
+        .apply(mock(JsonNode.class));
+
+    assertThat(resultFormatter, instanceOf(GcsBigQueryDenormalizedRecordFormatter.class));
+  }
+
+  @Test
+  void putStreamIntoUploaderMap_compareSchemas_expectedIsNotNullExistingIsNull() throws IOException {
+    final Map<AirbyteStreamNameNamespacePair, AbstractBigQueryUploader<?>> uploaderMap = new HashMap<>();
+    final String streamName = "stream_name";
+    final String nameSpace = "name_space";
+    final Table tableMock = mock(Table.class);
+    final Schema schemaMock = mock(Schema.class);
+    final TableDefinition tableDefinitionMock = mock(TableDefinition.class);
+
+    when(uploaderConfigMock.getConfigStream()).thenReturn(configuredStreamMock);
```

Comment:
it looks like this setup code is repeated a lot, can we extract it into a method?

Reply:
I extract some part but setup is not always same, it has some differences